source/manual/alerts/rabbitmq-no-consumers-listening.html.md

```diff
@@ -4,75 +4,80 @@ title: 'RabbitMQ: No consumers listening to queue'
 parent: "/manual.html"
 layout: manual_layout
 section: Icinga alerts
-last_reviewed_on: 2019-12-05
+last_reviewed_on: 2020-08-17
 review_in: 6 months
 ---
 
-[Read more about how we use RabbitMQ][rabbitmq]
+Publishing API [sends a heartbeat message][heartbeat_messages] every minute, so in principle
+there should always be at least one consumer listening.
 
-We run a check that there is at least one non-idle consumer for a named
-RabbitMQ queue. (For example, for the `email_alert_service` queue.) The queue
-name should indicate the app responsible for consuming the queue.
+This alert indicates that no consumer was found listening to the queue within
+the last 5 minutes.
 
-The Icinga check is performed by connecting to RabbitMQ's admin API, so the
-information given is from Rabbit's point of view.
+A successful check returns the time of the most recent activity on the queue.
 
-Since we send heartbeats every minute (from each instance of the Publishing
-API) at least one of the consumers for each queue should be active every
-minute. To be conservative, the check checks that at least one consumer of the
-queue is running and has been active within the last 5 minutes.
-
-If the check succeeds, it will return the most recent time at which activity
-happened on the queue.
-
-If the check fails due to no consumers having been active recently, it will
-report a critical failure, giving the most recent idle time, and the idle
-times of each consumer connected to RabbitMQ:
+When the check fails due to idle consumers, it reports a critical failure indicating
+the most recent idle time, and the idle times of each consumer connected to RabbitMQ:
 
 ```
 CRITICAL: No activity for X seconds: idle times are [X, Y, Z]
 ```
 
-If the check fails due to there being no consumers connected to the queue, it
-will report a critical failure:
+When the check fails because no consumers are listening to the queue, it
+reports a critical failure:
 
 ```
 CRITICAL: "No consumers listening to queue"
 ```
 
+## How this check works
+
+Icinga connects to RabbitMQ's admin API to check that at least one consumer is running
+for a given RabbitMQ queue. The queue name indicates the application consuming the queue,
+e.g. the `email_alert_service` queue.
+
 ## Consequences of idle consumers
 
-If consumers are idle, messages sent to the queue will not be being processed.
-This means that the actions which are meant to happen in response to the
-messages will not happen. For example, email alerts about updated content will
-not be sent if the `email_alert_sevice` queue isn't running.
+If consumers are idle, messages sent to the queue are not be processed and their associated
+actions are not triggered. E.g. If the `email_alert_service` queue isn't running, content updates
+don't trigger any email alerts.
```

Comment:
I'm not sure we've quite the terminology correct on the end of this paragraph. We're suggesting that the `email_alert_service` queue isn't running but I'd of thought the problem is not whether the queue exists or not but more whether the application connected to the queue is operational?

Do these checks also serve to determine whether the queue exists or not or is that managed elsewhere?

Reply:
@kevindew Yeah you're right, I think what is meant here is "idle consumers" rather than "queue not running".